akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -35,3 +43,29 @@ trait SinkQueue[T] {
    */
   def pull(): Future[Option[T]]
 }
+
+sealed trait StreamCallbackStatus[+T]
```

Comment:
please make it an `abstract class` instead of a `trait` because this package is to be consumed by Scala and Java alike